browser/resources/brave_new_tab_page_refresh/components/common/visibility_tracker.tsx

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+interface Props {
+  onVisible: () => void
+  rootMargin?: string
+}
+
+export function VisibilityTracker(props: Props) {
```

Comment:
We already have `useVisible` in `components/common/useVisible.ts` - doesn't news already use that?